inference-engine/src/gna_plugin/backend/dnn_components.cpp

```diff
@@ -8,32 +8,60 @@
 #include <ie_common.h>
 #include <legacy/ie_layers.h>
 #include <iomanip>
+#include <details/caseless.hpp>
+#include <layers/gna_copy_layer.hpp>
 #include "backend/dnn_types.h"
 
 #include "dnn_components.hpp"
 
 using namespace GNAPluginNS;
+using namespace GNAPluginNS::backend;
 
-intel_dnn_component_t & backend::DnnComponents::addComponent(const std::string layerName, const std::string layerMetaType) {
-    components.emplace_back(layerName, intel_dnn_component_t());
-    auto &currentComponent = components.back().second;
+intel_dnn_component_t & DnnComponents::addComponent(const std::string layerName, const std::string layerMetaType) {
+    auto isDelayed = InferenceEngine::details::CaselessEq<std::string>()(layerMetaType, DelayedCopyLayerName);
+    delayedOperations += isDelayed ? 1 : 0;
+    components.emplace_back(DnnComponentExtra{layerName, {}, isDelayed});
+    auto &currentComponent = components.back().dnnComponent;
 #ifdef PLOT
-    currentComponent.original_layer_name = components.back().first.c_str();
-    std::cout << "IR layer : " << std::left << std::setw(20) << layerName << " " << layerMetaType << "_" << components.size() - 1 << std::endl;
+    currentComponent.original_layer_name = components.back().name.c_str();
 #endif
+    int execOrder = 0;
+    if (!isDelayed) {
+        execOrder = components.size() - 1 - delayedOperations;
+    } else {
+        // todo: not perfect - propose to create mapping table that will be printed out by extra request
+        execOrder = -delayedOperations;
+    }
+
+    gnalog() << "IR layer : " << std::left << std::setw(20) << layerName << " " << layerMetaType << "_" << execOrder << std::endl;
     return currentComponent;
 }
 
-intel_dnn_component_t * backend::DnnComponents::findComponent(InferenceEngine::CNNLayerPtr __layer) {
+intel_dnn_component_t * DnnComponents::findComponent(InferenceEngine::CNNLayerPtr __layer) {
     auto component = std::find_if(begin(components),
                                   end(components),
                                   [&](storage_type ::value_type &comp) {
-                                      return comp.first == __layer->name;
+                                      return comp.name == __layer->name;
                                   });
     // check for generic prev layer
     if (component != components.end()) {
-        return &component->second;
+        return &component->dnnComponent;
     }
 
     return nullptr;
 }
+
+std::vector<intel_dnn_component_t> DnnComponents::getExecutionOrder() {
+    std::vector<intel_dnn_component_t> result(components.size());
+
+    uint32_t direct_id = 0;
+    uint32_t delayed_id = components.size() - delayedOperations;
```

Comment:
Shouldn't it be components.size() - 1 - delayedOperations?

Reply:
i think this is correct, direct_id will start from 0 and go up to number_of_not_delayed - 1, while delayed_id will start from number_of_not_delayed and go up to components.size() - 1 .

Reply:
You are correct.